src/Umbraco.Web.BackOffice/Filters/FileUploadCleanupFilterAttribute.cs

```diff
@@ -0,0 +1,163 @@
+                                var dir = Path.GetDirectoryName(f.TempFilePath);
+                                if (tempFolders.Contains(dir) == false)
+                                {
+                                    tempFolders.Add(dir);
+                                }
+
+                                try
+                                {
+                                    File.Delete(f.TempFilePath);
+                                }
+                                catch (Exception ex)
+                                {
+                                    _logger.Error<FileUploadCleanupFilterAttribute>(ex,
+                                        "Could not delete temp file {FileName}", f.TempFilePath);
+                                }
+                            }
+                        }
+                    }
+                }
+                else
+                {
+                    if (context == null)
+                    {
+                        _logger.Warn<FileUploadCleanupFilterAttribute>("The context is null!!??");
+                        return Task.CompletedTask;
+                    }
+
+                    if (context.Result == null)
+                    {
+                        _logger.Warn<FileUploadCleanupFilterAttribute>(
+                            "The context.Result is null!!??");
+                        return Task.CompletedTask;
+                    }
+
+                    if(!(context.Result is ObjectResult objectResult))
+                    {
+                        _logger.Warn<FileUploadCleanupFilterAttribute>(
+                            "Could not acquire context.Result as ObjectResult");
+                        return Task.CompletedTask;
+                    }
+
+                    if (objectResult.Value is IHaveUploadedFiles uploadedFiles)
+                    {
+                        if (uploadedFiles.UploadedFiles != null)
+                        {
+                            //cleanup any files associated
+                            foreach (var f in uploadedFiles.UploadedFiles)
+                            {
+                                if (f.TempFilePath.IsNullOrWhiteSpace() == false)
+                                {
+                                    //track all temp folders so we can remove old files afterwards
+                                    var dir = Path.GetDirectoryName(f.TempFilePath);
+                                    if (tempFolders.Contains(dir) == false)
+                                    {
+                                        tempFolders.Add(dir);
+                                    }
+
+                                    _logger.Debug<FileUploadCleanupFilterAttribute>(
+                                        "Removing temp file {FileName}", f.TempFilePath);
+
+                                    try
+                                    {
+                                        File.Delete(f.TempFilePath);
```

Comment:
Ends up quite nested but I think sometimes that is clearer than returning after the opposite case